src/workerd/jsg/README.md

```diff
@@ -640,6 +640,23 @@ the necessary mapping between the C++ struct and the JavaScript object. Only the
 properties listed in the macro are mapped. (In this case, `onlyInternal` is not
 included in the JavaScript object.)
 
+If the struct has a validate() method, it is called when the struct is unwrapped from v8.
+This is an opportunity for it to throw a TypeError based on some custom logic.
+
+```cpp
+struct ValdiatingFoo {
+  kj::String abc;
+
+  void validate() {
+    JSG_REQUIRE(abc.size() != 0, TypeError, "Field 'abc' had no length in 'ValidatingFoo'.");
```

Comment:
suggestion: might be worthwhile expanding the example a bit to show how to incorporate a compat flag check into the validation... but given that JSG doesn't directly know about compat flags, I wouldn't say it is critical

Reply:
I added a comment explaing that you have to use the direct API. It might make sense to force passing a reference to a jsg lock, but I wasn't sure how to make one in `tryUnwrap`.

Reply:
I beliefve you can get the current `jsg::Lock&` from the `info.GetIsolate()` like `jsg::Lock::from(info.GetIsolate())` etc

Reply:
do we want `validate` to have to take in a `jsg::Lock`?

Reply:
I think it would be a good idea.